libyara/object.c

```diff
@@ -56,7 +56,7 @@ int yr_object_create(
   YR_OBJECT* obj;
   size_t object_size = 0;
 
-  assert(parent != NULL || object != NULL);
+  assert(identifier != NULL || parent != NULL || object != NULL);
```

Comment:
The identifier should be checked in a separate `assert`. Otherwise, if identifier is not null, the other part of the condition (`parent != NULL || object != NULL`) won't be taken into account, regardless of the values of `parent` and `object`

Reply:
oops. Thanks for that.